syntax = "proto3";
import "google/protobuf/empty.proto";
import "component_proto/scene_comp.proto";
option cc_generic_services = true;

message CentreEnterSceneRequest
{
  SceneInfo scene_info = 1;
}

message CentreLeaveSceneRequest
{
}

message CentreLeaveSceneAsyncSavePlayerCompleteRequest
{
}

service CentreScenePlayerService 
{
  rpc EnterScene (CentreEnterSceneRequest) returns (google.protobuf.Empty);
  rpc LeaveScene (CentreLeaveSceneRequest) returns (google.protobuf.Empty);
  rpc LeaveSceneAsyncSavePlayerComplete (CentreLeaveSceneAsyncSavePlayerCompleteRequest) returns (google.protobuf.Empty);
}